packages/rum-core/src/boot/rumPublicApi.ts

```diff
@@ -201,40 +217,47 @@ export function makeRumPublicApi(
     }),
 
     setUser: monitor((newUser: User) => {
-      const sanitizedUser = sanitizeUser(newUser)
-      if (sanitizedUser) {
-        user = sanitizedUser
-      } else {
+      if (typeof newUser !== 'object' || !newUser) {
         display.error('Unsupported user:', newUser)
+      } else {
+        userContextManager.setContext(sanitizeUser(newUser as Context))
       }
     }),
 
+    getUser: monitor(userContextManager.getContext),
+
+    setUserProperty: monitor(userContextManager.setContextProperty),
+
+    removeUserProperty: monitor(userContextManager.removeContextProperty),
+
+    clearUser: monitor(userContextManager.clearContext),
+
     removeUser: monitor(() => {
-      user = {}
+      userContextManager.clearContext()
     }),
 
+    addUserAttribute: monitor(userContextManager.setContextProperty),
+
+    removeUserAttribute: monitor(userContextManager.removeContextProperty),
+
     startView,
 
     startSessionReplayRecording: monitor(recorderApi.start),
     stopSessionReplayRecording: monitor(recorderApi.stop),
   })
   return rumPublicApi
 
-  function sanitizeUser(newUser: unknown) {
-    if (typeof newUser !== 'object' || !newUser) {
-      return
-    }
-    const result = deepClone(newUser as Context)
-    if ('id' in result) {
-      result.id = String(result.id)
+  function sanitizeUser(newUser: Context) {
+    if ('id' in newUser) {
+      newUser.id = String(newUser.id)
     }
-    if ('name' in result) {
-      result.name = String(result.name)
+    if ('name' in newUser) {
+      newUser.name = String(newUser.name)
     }
-    if ('email' in result) {
-      result.email = String(result.email)
+    if ('email' in newUser) {
+      newUser.email = String(newUser.email)
     }
-    return result
+    return newUser
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏you should not mutate the variable coming from outside. Also, it could be nice to sanitize values set with `setUserProperty` as well?